apps/io-wallet-user-func/src/email-provider.ts

```diff
@@ -0,0 +1,55 @@
+import emailTemplate from "@/templates/wallet-instance-activation/index.html";
+import {
+  getMailerTransporter,
+  sendMail,
+} from "@pagopa/io-functions-commons/dist/src/mailer";
+import * as RTE from "fp-ts/lib/ReaderTaskEither";
+import * as TE from "fp-ts/lib/TaskEither";
+import { pipe } from "fp-ts/lib/function";
+
+import { MailConfig } from "./app/config";
+
+export const sendEmail: (
+  fiscalCode: string,
+) => RTE.ReaderTaskEither<{ mail: MailConfig }, Error, void> =
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  (fiscalCode) => (configs) =>
+    pipe(
+      TE.tryCatch(
+        async () => {
+          if (!configs.mail.enabled) {
+            return;
+          }
+          const userEmail = ""; // to do - get the email by fiscalCode
+
+          await sendMail(
+            getMailerTransporter({
+              MAIL_FROM: configs.mail.mailhogHost?.length
+                ? configs.mail.mailSender
+                : userEmail,
```

Comment:
why do you check the host length?

Reply:
I didn't want to do that, but the `getMailerTransporter()` function expects exactly `undefined`. So, even if when fetching the configurations I return an empty string (file `config.ts`, line `63`) because certain properties have not been configured, the easiest way to understand if the configuration exists or not is the length check.

A small note. If I try to do something like this:

```
readFromEnvironment("MailSender"),
RE.orElse(() => RE.right(undefined)),
```

The linter shows an error, because the `readFromEnvironment()` function expects an `Either<Error, string>`, not an `Either<Error, undefined>`. The solution could be to change the signature of the `readFromEnvironment()` function in the `io-wallet-common` package, but I don't understand what the side effects might be.

Reply:
See previous comment. You don't have to do all these checks

Reply:
Makes sense! I proceed by removing the `MailhogHost` environment variable, because this is an attribute not required to configure MailUp as a mail service. This solves a couple of comments, as only the username and secret are needed.